### ISSUE TITLE
Function naming is mixedup

### ISSUE THREAD

**Issue by EricSch** [rollup/rollup#2171]
I have a specific vase with an module, where function names are resolved wrong.
```
function toDataURL(content) {
    return DATA_URL_PREFIX + content;
}

Workbook.prototype.toDataURL = function toDataURL$1 () {
      var zip = this.toZIP();
      return zip.generateAsync ? zip.generateAsync(DATA_URL_OPTIONS).then(toDataURL) : toDataURL(zip.generate(DATA_URL_OPTIONS));
 };

```
That will be bundled to:
```
function toDataURL$1(content) {
    return DATA_URL_PREFIX + content;
}

Workbook.prototype.toDataURL = function toDataURL$1 () {
      var zip = this.toZIP();
      return zip.generateAsync ? zip.generateAsync(DATA_URL_OPTIONS).then(toDataURL$1) : toDataURL$1(zip.generate(DATA_URL_OPTIONS));
 };
```
Means the function is called recursive (Line with return zip.....).
I made an repl, but can't share the URL. Too much code.

So here the steps to replicate:

1. Get the package `@progress/kendo-ooxml`
2. In main.js:
```
/* TREE-SHAKING */
import { Workbook } from './ooxml.js';
var b = new Workbook()
b.toZIP();
```
3. Add as module ooxml.js
4. In the IIFE output, you see, that the function toDataURL is wrong. The 'toDataURL'-Function in the "return" line should be the global function.
   ```
   toDataURL() {
          const zip = this.toZIP();

          return zip.generateAsync ? zip.generateAsync(DATA_URL_OPTIONS).then(toDataURL) : toDataURL(zip.generate(DATA_URL_OPTIONS));
      }
```
Hope this helps
Thanks

**Comment by guybedford** [comment]
Thanks for posting the issue - this does seem like it could be an issue with the deshadowing. I ran a quick test in https://rollupjs.org/repl?version=0.58.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMG5hbWUlMjBhcyUyMG4lMkMlMjBuYW1lJTI0MSUyMGFzJTIwciUyMCU3RCUyMGZyb20lMjAnLiUyRm1hdGhzLmpzJyUzQiU1Q24lNUNudmFyJTIwbmFtZSUyMCUzRCUyMCdhc2RmJyUzQiU1Q252YXIlMjBuYW1lJTI0MSUyMCUzRCUyMCdhbm90aGVyJyUzQiU1Q24lNUNuY29uc29sZS5sb2cobiUyMCUyQiUyMG5hbWUlMjAlMkIlMjBuYW1lJTI0MSUyMCUyQiUyMHIpJTNCJTIyJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMm1hdGhzLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMHZhciUyMG5hbWUlMjAlM0QlMjA1JTNCJTVDbmV4cG9ydCUyMHZhciUyMG5hbWUlMjQxJTIwJTNEJTIwMTAlM0IlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=, but couldn't get any of these cases to show up.

The package `@progress/kendo-ooxml` isn't actually available on npm - it would help if you can share a simple replication here to look into it further.

**Comment by guybedford** [comment]
I finally got a replication of this here - https://rollupjs.org/repl?version=0.58.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMG5hbWUlMjBhcyUyMG4lMkMlMjByJTIwJTdEJTIwZnJvbSUyMCcuJTJGbWF0aHMuanMnJTNCJTVDbiU1Q25uKCklM0IlNUNucigpJTNCJTIyJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMm1hdGhzLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QiUyMG5hbWUlMjBhcyUyMHIlMjAlN0QlMjBmcm9tJTIwJy4lMkZhbm90aGVyLmpzJyUzQiU1Q24lNUNuZXhwb3J0JTIwZnVuY3Rpb24lMjBuYW1lJTIwKCklMjAlN0IlMjBjb25zb2xlLmxvZygncmVuYW1lZCcpJTIwJTdEJTVDbiU1Q252YXIlMjBhJTIwJTNEJTIwZnVuY3Rpb24lMjBuYW1lJTI0MSUyMCgpJTIwJTdCJTVDbiU1Q3RyZXR1cm4lMjBuYW1lKCkudGhlbihuYW1lKSUzQiU1Q24lN0QlNUNuJTVDbmEoKSUzQiUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJhbm90aGVyLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGZ1bmN0aW9uJTIwbmFtZSUyMCgpJTIwJTdCJTIwY29uc29sZS5sb2coJ2Fub3RoZXInKSUyMCU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

The issue seems to be that the function name itself is not taken into account during the deshadowing check process.

**Comment by EricSch** [comment]
@guybedford Thanks for the sample. The package should be available for the public. https://www.npmjs.com/package/@progress/kendo-ooxml.
But you got my problem, so thanks.

**Comment by lukastaegert** [comment]
Finally fixed as of Rollup@1.2.0:
https://rollupjs.org/repl?version=1.2.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMG5hbWUlMjBhcyUyMG4lMkMlMjByJTIwJTdEJTIwZnJvbSUyMCcuJTJGbWF0aHMuanMnJTNCJTVDbiU1Q25uKCklM0IlNUNucigpJTNCJTIyJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMm1hdGhzLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QiUyMG5hbWUlMjBhcyUyMHIlMjAlN0QlMjBmcm9tJTIwJy4lMkZhbm90aGVyLmpzJyUzQiU1Q24lNUNuZXhwb3J0JTIwZnVuY3Rpb24lMjBuYW1lJTIwKCklMjAlN0IlMjBjb25zb2xlLmxvZygncmVuYW1lZCcpJTIwJTdEJTVDbiU1Q252YXIlMjBhJTIwJTNEJTIwZnVuY3Rpb24lMjBuYW1lJTI0MSUyMCgpJTIwJTdCJTVDbiU1Q3RyZXR1cm4lMjBuYW1lKCkudGhlbihuYW1lKSUzQiU1Q24lN0QlNUNuJTVDbmEoKSUzQiUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJhbm90aGVyLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGZ1bmN0aW9uJTIwbmFtZSUyMCgpJTIwJTdCJTIwY29uc29sZS5sb2coJ2Fub3RoZXInKSUyMCU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==
